### PR TITLE
Update deployment_guide.md

### PR DESCRIPTION
**Pull request by noklam**
## Description

The deployment doc is missing airflow in both text & the flowchart.

## Development notes

See the chart below.
image

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1771"></a>

### CODE REVIEW

**Comment by noklam** [comment]
flowchart TD
    A{Can your Kedro pipeline run on a single machine?} -- YES --> B[Consult the single-machine deployment guide];
    B --> C{Do you have Docker on your machine?};
    C -- YES --> D[Use a container-based approach];
    C -- NO --> E[Use the CLI or package mode];
    A -- NO --> F[Consult the distributed deployment guide];
    F --> G["What distributed platform are you using?<br/><br/>Check out the guides for:<br/><br/><li>Argo</li><li>Prefect</li><li>Kubeflow Pipelines</li><li>AWS Batch</li><li>Databricks</li><li>Dask</li>"];
    style G text-align:left
    H["Does (part of) your pipeline integrate with Amazon SageMaker?<br/><br/>Read the SageMaker integration guide"];
    style H text-align:left

**Comment by noklam** [comment]
```{mermaid}
flowchart TD
    A{Can your Kedro pipeline run on a single machine?} -- YES --> B[Consult the single-machine deployment guide];
    B --> C{Do you have Docker on your machine?};
    C -- YES --> D[Use a container-based approach];
    C -- NO --> E[Use the CLI or package mode];
    A -- NO --> F[Consult the distributed deployment guide];
    F --> G["What distributed platform are you using?<br/><br/>Check out the guides for:<br/><br/>
    <li>Argo</li>
    <li>Prefect</li>
    <li>Kubeflow Pipelines</li>
    <li>AWS Batch</li>
    <li>Databricks</li>
    <li>Dask</li>"];
    style G text-align:left
    H["Does (part of) your pipeline integrate with Amazon SageMaker?<br/><br/>Read the SageMaker integration guide"];
    style H text-align:left
```

**Review by datajoely — CHANGES_REQUESTED** [review]

File: docs/source/deployment/deployment_guide.md [hunk 1]
```diff
@@ -28,7 +29,7 @@ flowchart TD
     C -- YES --> D[Use a container-based approach];
     C -- NO --> E[Use the CLI or package mode];
     A -- NO --> F[Consult the distributed deployment guide];
-    F --> G["What distributed platform are you using?<br/><br/>Check out the guides for:<br/><br/><li>Argo</li><li>Prefect</li><li>Kubeflow Pipelines</li><li>AWS Batch</li><li>Databricks</li><li>Dask</li>"];
+    F --> G["What distributed platform are you using?<br/><br/>Check out the guides for:<br/><br/><li>Airflow</li><li>Argo</li><li>Prefect</li><li>Kubeflow Pipelines</li><li>AWS Batch</li><li>Databricks</li><li>Dask</li>"];
     style G text-align:left
     H["Does (part of) your pipeline integrate with Amazon SageMaker?<br/><br/>Read the SageMaker integration guide"];
```
[the remark this answers is not in the document]
Comment (noklam) [reply docs/source/deployment/deployment_guide.md:34]: @datajoely I put up a screenshot on the description, does it looks good to you?

**Review by yetudada — APPROVED** [review]
Thanks for this @noklam 🥳

**Pull request merged**